### PR TITLE
Add ALETH42 keyboard

### PR DESCRIPTION
**Pull request by monksoffunk**
## Description

This PR adds the ALETH42 keyboard.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/aleth42/keymaps/default/keymap.c [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+    QWERTY = SAFE_RANGE,
+    LOWER,
+    RAISE,
+    ADJUST,
+};
+
+#define L_QWERTY 0
+#define L_LOWER 1 << _LOWER
+#define L_RAISE 1 << _RAISE
+#define L_ADJUST 1 << _ADJUST
```
Comment (drashna) [inline keyboards/aleth42/keymaps/default/keymap.c:37]: I'm not really a fan of this.

Comment (monksoffunk) [reply keyboards/aleth42/keymaps/default/keymap.c:37]: Is this the better?

#define L_QWERTY 0
#define L_LOWER 2
#define L_RAISE 4
#define L_ADJUST 8

Comment (monksoffunk) [reply keyboards/aleth42/keymaps/default/keymap.c:37]: @drashna  How should I do?

File: keyboards/aleth42/keymaps/default/keymap.c [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+        KC_MUTE, KC_F1,   KC_F2, KC_F3, KC_F4, KC_F5, KC_F6, KC_F7, KC_F8, KC_F9, KC_F10, KC_F11,
+		_______, BL_TOGG, BL_STEP, BL_INC,  AG_NORM,  RGB_TOG, RGB_HUI, AG_SWAP, RGB_SAI, RGB_VAI,  KC_F12,
+		KC_CAPS, RESET,   BL_BRTG, BL_DEC,  _______,  RGB_MOD, RGB_HUD, _______, RGB_SAD, RGB_VAD,  _______,
+		KC_SLEP, _______, _______, _______, _______, _______, _______, _______
+		),
+};
+
+void encoder_update_user(uint8_t index, bool clockwise) {
+    if (index == 0) { /* Left encoder */
+        switch (layer_state) {
```
[the remark this answers is not in the document]
Comment (monksoffunk) [reply keyboards/aleth42/keymaps/default/keymap.c]: Thanks!
I added the commit.

File: keyboards/aleth42/keymaps/default/keymap.c (hunk 1 above)
Comment (drashna) [reply keyboards/aleth42/keymaps/default/keymap.c]: Sorry, it would be best to just use the layer numbers, and use the `get_highest_layer(state)`.

**Review by drashna — APPROVED** [review]

File: keyboards/aleth42/keymaps/default/keymap.c [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+void encoder_update_user(uint8_t index, bool clockwise) {
+    if (index == 0) { /* Left encoder */
+        switch (get_highest_layer(layer_state)) {
+            case _QWERTY:
+                if (clockwise) {
+                    tap_code(KC_TAB);
+                } else {
+                    register_code(KC_LSFT);
+                    tap_code(KC_TAB);
+                    unregister_code(KC_LSFT);
```
Comment (fauxpark) [inline keyboards/aleth42/keymaps/default/keymap.c:117]:
```suggestion
                    tap_code16(S(KC_TAB));
```

File: keyboards/aleth42/keymaps/default/keymap.c [hunk 4]
```diff
@@ -0,0 +1,151 @@
...
+                }
+                break;
+            case _RAISE:
+                if (clockwise) {
+                //    tap_code(KC_VOLU);
+                    if(keymap_config.swap_lalt_lgui==false){
+                        tap_code(KC_LANG2);
+                    }else {
+                        SEND_STRING(SS_LALT("`"));
```
Comment (fauxpark) [inline keyboards/aleth42/keymaps/default/keymap.c:126]:
```suggestion
                        tap_code16(A(KC_GRV));
```

File: keyboards/aleth42/keymaps/default/keymap.c [hunk 5]
```diff
@@ -0,0 +1,151 @@
...
+                    }
+                } else {
+                    if(keymap_config.swap_lalt_lgui==false){
+                    tap_code(KC_LANG1);
+                    } else {
+                        SEND_STRING(SS_LALT("`"));
```
Comment (fauxpark) [inline keyboards/aleth42/keymaps/default/keymap.c:132]:
```suggestion
                        tap_code16(A(KC_GRV));
```

File: keyboards/aleth42/keymaps/default/readme.md [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+  Default Layer
+      ,-----------------------------------------------------------.
+      | Esc|  Q |  W |  E |  R |  T |  Y |  U |  I |  O |  P | BS |
+      |-----------------------------------------------------------|
+      | Tab |  A |  S |  D |  F |  G |  H |  J |  K |  L | Ent    |
+      |-----------------------------------------------------------|
+      | LSft   |  Z |  X |  C |  V |  B |  N |  M |  , |  . |fn(/)|
+      |-----------------------------------------------------------|
+      | Lctl | LAlt| LGui|  spc fn0  |  spc fn1    |RGui|RAlt|RCtl|
+      `-----------------------------------------------------------'
```
Comment (fauxpark) [inline keyboards/aleth42/keymaps/default/readme.md:12]:
```suggestion
Default Layer
    ,-----------------------------------------------------------.
    | Esc|  Q |  W |  E |  R |  T |  Y |  U |  I |  O |  P | BS |
    |-----------------------------------------------------------|
    | Tab |  A |  S |  D |  F |  G |  H |  J |  K |  L | Ent    |
    |-----------------------------------------------------------|
    | LSft   |  Z |  X |  C |  V |  B |  N |  M |  , |  . |fn(/)|
    |-----------------------------------------------------------|
    | Lctl | LAlt| LGui|  spc fn0  |  spc fn1    |RGui|RAlt|RCtl|
    `-----------------------------------------------------------'
```

File: keyboards/aleth42/keymaps/default/readme.md [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+  Lower Layer = fn0
+      ,-----------------------------------------------------------.
+      |  ~ |  ! |  @ |  # |  $ |  % |  ^ |  & |  * |  ( |  ) | Del|
+      |-----------------------------------------------------------|
+      |        |    |    |    |    |     | _ | + | { | } |  Pipe  |
+      |-----------------------------------------------------------|
+      |          |    |    |    |    |    |   | ;  | '  | Up |    |
+      |-----------------------------------------------------------|
+      |     |    |     |           |             |Left|Down|Right |
+      `-----------------------------------------------------------'
```
Comment (fauxpark) [inline keyboards/aleth42/keymaps/default/readme.md:23]:
```suggestion
Lower Layer = fn0
    ,-----------------------------------------------------------.
    |  ~ |  ! |  @ |  # |  $ |  % |  ^ |  & |  * |  ( |  ) | Del|
    |-----------------------------------------------------------|
    |        |    |    |    |    |     | _ | + | { | } |  Pipe  |
    |-----------------------------------------------------------|
    |          |    |    |    |    |    |   | ;  | '  | Up |    |
    |-----------------------------------------------------------|
    |     |    |     |           |             |Left|Down|Right |
    `-----------------------------------------------------------'
```

File: keyboards/aleth42/keymaps/default/readme.md [hunk 8]
```diff
@@ -0,0 +1,46 @@
...
+	Raise Layer = fn1
+      ,-----------------------------------------------------------.
+      |  ` |  1 |  2 |  3 |  4 |  5 |  6 |  7 |  8 |  9 |  0 | Del|
+      |-----------------------------------------------------------|
+      |  F1   | F2  | F3 | F4 | F5 |  F6 | - | = | [ | ] |  Pipe  |
+      |-----------------------------------------------------------|
+      |    F7    | F8 | F9 | F10| F11| F12|  \ |  \ |   |    |    |
+      |-----------------------------------------------------------|
+      |     |    |     |           |             |     |    |     |
+      `-----------------------------------------------------------'
```
Comment (fauxpark) [inline keyboards/aleth42/keymaps/default/readme.md:34]:
```suggestion
Raise Layer = fn1
    ,-----------------------------------------------------------.
    |  ` |  1 |  2 |  3 |  4 |  5 |  6 |  7 |  8 |  9 |  0 | Del|
    |-----------------------------------------------------------|
    |  F1   | F2  | F3 | F4 | F5 |  F6 | - | = | [ | ] |  Pipe  |
    |-----------------------------------------------------------|
    |    F7    | F8 | F9 | F10| F11| F12|  \ |  \ |   |    |    |
    |-----------------------------------------------------------|
    |     |    |     |           |             |     |    |     |
    `-----------------------------------------------------------'
```

File: keyboards/aleth42/keymaps/default/readme.md [hunk 9]
```diff
@@ -0,0 +1,46 @@
...
+	Adjust Layer = fn
+      ,-----------------------------------------------------------.
+      |Mute| F1 | F2 | F3 | F4 | F5 | F6 | F7 | F8 | F9 | F10|F11 |
+      |-----------------------------------------------------------|
+      |       |BTOG |BSTP|BINC| MAC|RGBTOG|HUI|WIN|SAI|VAI|  F12  |
+      |-----------------------------------------------------------|
+      |   Caps   |Reset|BBRE|BDEC|    |RMOD|HUD |   |SAD|VAD |    |
+      |-----------------------------------------------------------|
+      |SLEEP|    |     |           |             |     |    |     |
+      `-----------------------------------------------------------'
```
Comment (fauxpark) [inline keyboards/aleth42/keymaps/default/readme.md:45]:
```suggestion
Adjust Layer = fn
    ,-----------------------------------------------------------.
    |Mute| F1 | F2 | F3 | F4 | F5 | F6 | F7 | F8 | F9 | F10|F11 |
    |-----------------------------------------------------------|
    |       |BTOG |BSTP|BINC| MAC|RGBTOG|HUI|WIN|SAI|VAI|  F12  |
    |-----------------------------------------------------------|
    |   Caps   |Reset|BBRE|BDEC|    |RMOD|HUD |   |SAD|VAD |    |
    |-----------------------------------------------------------|
    |SLEEP|    |     |           |             |     |    |     |
    `-----------------------------------------------------------'
```

File: keyboards/aleth42/keymaps/default/config.h [hunk 10]
```diff
@@ -0,0 +1,22 @@
+/* Copyright 2020 monksoffunk
```
Comment (fauxpark) [inline keyboards/aleth42/keymaps/default/config.h:1]: This file can be deleted.

File: keyboards/aleth42/keymaps/via/readme.md [hunk 11]
```diff
@@ -0,0 +1,50 @@
...
+  Default Layer
+      ,-----------------------------------------------------------.
+      | Esc|  Q |  W |  E |  R |  T |  Y |  U |  I |  O |  P | BS |
+      |-----------------------------------------------------------|
+      | Tab*|  A |  S |  D |  F |  G |  H |  J |  K |  L | Ent    |
+      |-----------------------------------------------------------|
+      | LSft   |  Z |  X |  C |  V |  B |  N |  M |  , |  . |fn(/)|
+      |-----------------------------------------------------------|
+      | Esc* | LAlt| LGui|  spc fn0  |  spc fn1    |RGui|RAlt|RCtl|
+      `-----------------------------------------------------------'
```
Comment (fauxpark) [inline keyboards/aleth42/keymaps/via/readme.md:12]:
```suggestion
Default Layer
    ,-----------------------------------------------------------.
    | Esc|  Q |  W |  E |  R |  T |  Y |  U |  I |  O |  P | BS |
    |-----------------------------------------------------------|
    | Tab*|  A |  S |  D |  F |  G |  H |  J |  K |  L | Ent    |
    |-----------------------------------------------------------|
    | LSft   |  Z |  X |  C |  V |  B |  N |  M |  , |  . |fn(/)|
    |-----------------------------------------------------------|
    | Esc* | LAlt| LGui|  spc fn0  |  spc fn1    |RGui|RAlt|RCtl|
    `-----------------------------------------------------------'
```

File: keyboards/aleth42/keymaps/via/readme.md [hunk 12]
```diff
@@ -0,0 +1,50 @@
...
+
+      Tab* would activate Left Control when held, and send Tab when tapped.
+      Esc* would activate Left Control when held, and send Escape when tapped.
+      Esc* is useful with a rotary encoder @ upper-left Esc location.
```
Comment (fauxpark) [inline keyboards/aleth42/keymaps/via/readme.md:16]:
```suggestion
Tab* would activate Left Control when held, and send Tab when tapped.

Esc* would activate Left Control when held, and send Escape when tapped.

Esc* is useful with a rotary encoder @ upper-left Esc location.
```

File: keyboards/aleth42/keymaps/via/readme.md [hunk 13]
```diff
@@ -0,0 +1,50 @@
...
+	Adjust Layer = fn
+      ,-----------------------------------------------------------.
+      |Mute| F1 | F2 | F3 | F4 | F5 | F6 | F7 | F8 | F9 | F10|F11 |
+      |-----------------------------------------------------------|
+      |       |BTOG |BSTP|BINC| MAC|RGBTOG|HUI|WIN|SAI|VAI|  F12  |
+      |-----------------------------------------------------------|
+      |   Caps   |Reset|BBRE|BDEC|    |RMOD|HUD |   |SAD|VAD |    |
+      |-----------------------------------------------------------|
+      |SLEEP|    |     |           |             |     |    |     |
+      `-----------------------------------------------------------'
```
Comment (fauxpark) [inline keyboards/aleth42/keymaps/via/readme.md:49]:
```suggestion
Lower Layer = fn0
    ,-----------------------------------------------------------.
    |  ~ |  ! |  @ |  # |  $ |  % |  ^ |  & |  * |  ( |  ) | Del|
    |-----------------------------------------------------------|
    |        |    |    |    |    |     | _ | + | { | } |  Pipe  |
    |-----------------------------------------------------------|
    |          |    |    |    |    |    |   | ;  | '  | Up |    |
    |-----------------------------------------------------------|
    |     |    |     |           |             |Left|Down|Right |
    `-----------------------------------------------------------'

Raise Layer = fn1
    ,-----------------------------------------------------------.
    |  ` |  1 |  2 |  3 |  4 |  5 |  6 |  7 |  8 |  9 |  0 | Del|
    |-----------------------------------------------------------|
    |  F1   | F2  | F3 | F4 | F5 |  F6 | - | = | [ | ] |  Pipe  |
    |-----------------------------------------------------------|
    |    F7    | F8 | F9 | F10| F11| F12|  \ |  \ |   |    |    |
    |-----------------------------------------------------------|
    |     |    |     |           |             |     |    |     |
    `-----------------------------------------------------------'

Adjust Layer = fn
    ,-----------------------------------------------------------.
    |Mute| F1 | F2 | F3 | F4 | F5 | F6 | F7 | F8 | F9 | F10|F11 |
    |-----------------------------------------------------------|
    |       |BTOG |BSTP|BINC| MAC|RGBTOG|HUI|WIN|SAI|VAI|  F12  |
    |-----------------------------------------------------------|
    |   Caps   |Reset|BBRE|BDEC|    |RMOD|HUD |   |SAD|VAD |    |
    |-----------------------------------------------------------|
    |SLEEP|    |     |           |             |     |    |     |
    `-----------------------------------------------------------'
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/aleth42/keymaps/default/config.h (hunk 10 above)
Comment (monksoffunk) [reply keyboards/aleth42/keymaps/default/config.h]: deleted!

**Comment by monksoffunk** [comment]
@fauxpark Thanks for your suggestions.
I updated the via keymap files as your suggestions to default keymap.

File: keyboards/aleth42/keymaps/default/readme.md [hunk 14]
```diff
@@ -0,0 +1,46 @@
+# The default keymap for aleth42
+
+Default Layer
+    ,-----------------------------------------------------------.
```
Comment (fauxpark) [inline keyboards/aleth42/keymaps/default/readme.md:4]:
```suggestion

    ,-----------------------------------------------------------.
```

File: keyboards/aleth42/keymaps/default/readme.md [hunk 15]
```diff
@@ -0,0 +1,46 @@
...
+    |-----------------------------------------------------------|
+    | Tab |  A |  S |  D |  F |  G |  H |  J |  K |  L | Ent    |
+    |-----------------------------------------------------------|
+    | LSft   |  Z |  X |  C |  V |  B |  N |  M |  , |  . |fn(/)|
+    |-----------------------------------------------------------|
+    | Lctl | LAlt| LGui|  spc fn0  |  spc fn1    |RGui|RAlt|RCtl|
+    `-----------------------------------------------------------'
+
+Lower Layer = fn0
+    ,-----------------------------------------------------------.
```
Comment (fauxpark) [inline keyboards/aleth42/keymaps/default/readme.md:15]:
```suggestion

    ,-----------------------------------------------------------.
```

File: keyboards/aleth42/keymaps/default/readme.md [hunk 16]
```diff
@@ -0,0 +1,46 @@
...
+    |-----------------------------------------------------------|
+    |        |    |    |    |    |     | _ | + | { | } |  Pipe  |
+    |-----------------------------------------------------------|
+    |          |    |    |    |    |    |   | ;  | '  | Up |    |
+    |-----------------------------------------------------------|
+    |     |    |     |           |             |Left|Down|Right |
+    `-----------------------------------------------------------'
+
+Raise Layer = fn1
+    ,-----------------------------------------------------------.
```
Comment (fauxpark) [inline keyboards/aleth42/keymaps/default/readme.md:26]:
```suggestion

    ,-----------------------------------------------------------.
```

File: keyboards/aleth42/keymaps/default/readme.md [hunk 17]
```diff
@@ -0,0 +1,46 @@
...
+    |-----------------------------------------------------------|
+    |  F1   | F2  | F3 | F4 | F5 |  F6 | - | = | [ | ] |  Pipe  |
+    |-----------------------------------------------------------|
+    |    F7    | F8 | F9 | F10| F11| F12|  \ |  \ |   |    |    |
+    |-----------------------------------------------------------|
+    |     |    |     |           |             |     |    |     |
+    `-----------------------------------------------------------'
+
+Adjust Layer = fn
+    ,-----------------------------------------------------------.
```
Comment (fauxpark) [inline keyboards/aleth42/keymaps/default/readme.md:37]:
```suggestion

    ,-----------------------------------------------------------.
```

File: keyboards/aleth42/keymaps/via/readme.md [hunk 18]
```diff
@@ -0,0 +1,52 @@
+# The VIA keymap for aleth42
+
+Default Layer
+    ,-----------------------------------------------------------.
```
Comment (fauxpark) [inline keyboards/aleth42/keymaps/via/readme.md:4]:
```suggestion

    ,-----------------------------------------------------------.
```

File: keyboards/aleth42/keymaps/via/readme.md [hunk 19]
```diff
@@ -0,0 +1,52 @@
...
+    `-----------------------------------------------------------'
+
+Tab* would activate Left Control when held, and send Tab when tapped.
+
+Esc* would activate Left Control when held, and send Escape when tapped.
+
+Esc* is useful with a rotary encoder @ upper-left Esc location.
+
+Lower Layer = fn0
+    ,-----------------------------------------------------------.
```
Comment (fauxpark) [inline keyboards/aleth42/keymaps/via/readme.md:21]:
```suggestion

    ,-----------------------------------------------------------.
```

File: keyboards/aleth42/keymaps/via/readme.md [hunk 20]
```diff
@@ -0,0 +1,52 @@
...
+    |-----------------------------------------------------------|
+    |        |    |    |    |    |     | _ | + | { | } |  Pipe  |
+    |-----------------------------------------------------------|
+    |          |    |    |    |    |    |   | ;  | '  | Up |    |
+    |-----------------------------------------------------------|
+    |     |    |     |           |             |Left|Down|Right |
+    `-----------------------------------------------------------'
+
+Raise Layer = fn1
+    ,-----------------------------------------------------------.
```
Comment (fauxpark) [inline keyboards/aleth42/keymaps/via/readme.md:32]:
```suggestion

    ,-----------------------------------------------------------.
```

File: keyboards/aleth42/keymaps/via/readme.md [hunk 21]
```diff
@@ -0,0 +1,52 @@
...
+    |-----------------------------------------------------------|
+    |  F1   | F2  | F3 | F4 | F5 |  F6 | - | = | [ | ] |  Pipe  |
+    |-----------------------------------------------------------|
+    |    F7    | F8 | F9 | F10| F11| F12|  \ |  \ |   |    |    |
+    |-----------------------------------------------------------|
+    |     |    |     |           |             |     |    |     |
+    `-----------------------------------------------------------'
+
+Adjust Layer = fn
+    ,-----------------------------------------------------------.
```
Comment (fauxpark) [inline keyboards/aleth42/keymaps/via/readme.md:43]:
```suggestion

    ,-----------------------------------------------------------.
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Review by fauxpark — APPROVED** [review]

**Review by zvecr — APPROVED** [review]

**Comment by zvecr** [comment]
Thanks!

**Comment by monksoffunk** [comment]
Thank you all!

**Pull request merged**